gpjax/likelihoods.py

```diff
@@ -85,6 +92,15 @@ def link_function(self, f: Float[Array, "..."]) -> tfd.Distribution:
         """
         raise NotImplementedError
 
+    def expected_log_likelihood(
+        self,
+        y: Float[Array, "N D"],
+        mu: Float[Array, "N D"],
+        sigma2: Float[Array, "N D"],
+    ):
+        log_prob = vmap(lambda f, y: self.link_function(f).log_prob(y))
+        return self.integrator(fun=log_prob, y=y, mean=mu, sigma2=sigma2, **self.dict())
+
```

Comment:
This is nice. Maybe I would either call this a variational expectation --- or reference this terminology in a docstring --- in keeping with the literature.